src/main/java/com/cb/controllers/ShopController.java

```diff
@@ -0,0 +1,84 @@
+
+import com.cb.dto.DefaultDTO;
+import com.cb.services.dbService.dbServiceImpl.ShopDBServiceImpl;
+import com.cb.services.dbService.iDbService.FighterDBService;
+import com.cb.services.mapService.iMapService.ShopService;
+import com.cb.services.mapService.iMapService.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpSession;
+
+
+@Controller
+public class ShopController {
+
+    @Autowired
+    ShopService shopService;
+
+    @Autowired
+    UserService userService;
+
+    @Autowired
+    FighterDBService fighterDBService;
+    @Autowired
+    ShopDBServiceImpl shopDBService;
+
+    @RequestMapping(method = RequestMethod.GET, value = "/shop")
+    public String getWeaponsTable(Model m, HttpServletRequest req) {
+
+        HttpSession userSession = req.getSession();
+        int userid = (int) userSession.getAttribute("id");
+        String username = (String) userSession.getAttribute("userName");
+        m.addAttribute("username", username);
+        int money = fighterDBService.getFighterMoneyByUserId(userid);
+        m.addAttribute("money", money);
+        DefaultDTO defaultDTO = shopService.getWeapons(userid);
+        m.addAttribute("weaponsListBL", defaultDTO);
+        return "shop";
+    }
+
+    @RequestMapping(method=RequestMethod.POST, value="fighter-weapon-map")
+    public String insertFighterWeapons(Model m, HttpServletRequest req,
+                                       @RequestParam(value="1", required = false) String s1,
+                                       @RequestParam(value = "2", required=false) String s2,
+                                       @RequestParam(value = "3", required=false) String s3,
+                                       @RequestParam(value = "4", required=false) String s4,
+                                       @RequestParam(value = "5", required=false) String s5,
+                                       @RequestParam(value="6", required = false) String s6,
+                                       @RequestParam(value = "7", required=false) String s7,
+                                       @RequestParam(value = "8", required=false) String s8,
+                                       @RequestParam(value = "9", required=false) String s9,
+                                       @RequestParam(value = "10", required=false) String s10,
+                                       @RequestParam(value = "hidden-field", required=false) int moneyLeft) {
+
+        HttpSession userSession = req.getSession();
+        int userid = (int) userSession.getAttribute("id");
+
+        String[] strings = {s1,s2,s3,s4,s5,s6,s7,s8,s9,s10};
+
```

Comment:
Here also, what are these variables?

Reply:
its because input fields are constructed when user clicks "Buy", we need to give each input field some unique name and match it to weapon..dont know how to explain better, just it works. Because first we get all inputs from request, some of them full and some of them are null. Then we filter out the null ones and only send forward the not null ones

Reply:
Let's discuss about this.